packages/react-portal/src/components/Portal/usePortalMountNode.ts

```diff
@@ -29,6 +30,8 @@ export const usePortalMountNode = (options: UsePortalMountNodeOptions) => {
     return newElement;
   }, [targetDocument, themeClassName, dir, options.disabled]);
 
+  (useKeyboardNavAttribute() as React.MutableRefObject<HTMLElement>).current = element!;
```

Comment:
why is this case necessary ?

Reply:
This case is necessary to ensure keyboard navigation outside the `FluentProvider` root. Without this any converged component that needs `react-portal` (like `react-menu`) would not have the keyboard navigation support.

Reply:
EDIT: `cast`

Reply:
Oh, now makes more sense 😅
The cast is necessary because by default references are immutable objects, and since this is a clear mutation, I'd need to cast it to it's mutable equivalent `React.MutableRefObject`

Reply:
why can't the hook return a mutable ref ? it should be possible to change the value of the ref

Reply:
As discussed offline, it seems non-mutable ref by default seems good until we get more scenarios that need explicit mutating of the ref